src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/DnsZone/New-ArmPrivateDnsZoneARecord.ps1

```diff
@@ -0,0 +1,44 @@
+function New-ArmPrivateDnsZoneARecord {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType("PDNSZARecord")]
+    Param(
+        [string]
+        [ValidatePattern('^(\[.*\]|)|([a-z0-9-_]{1,127})$')]
```

Comment:
A wildcard ('*' character) is permitted either as the single character in the name, or as the first label in the name. An empty value, or a single '@' character is permitted for record sets at the zone apex (except for CNAME record sets).

Reply:
Since it's a A record, it should also support `*.domain.com`, `@.domain.com` and `.domain.com`

Reply:
> A wildcard ('*' character) is permitted either as the single character in the name, or as the first label in the name. An empty value, or a single '@' character is permitted for record sets at the zone apex (except for CNAME record sets).

i think for the empty value and @ cases we should create separate PoshArm functions to reduce confusion